public/app/plugins/panel/timeseries/plugins/ExemplarMarker.tsx

```diff
@@ -122,13 +125,25 @@ export const ExemplarMarker = ({
   }, [setIsOpen]);
 
   const renderMarker = useCallback(() => {
-    //Put fields with links on the top
-    const fieldsWithLinks =
-      dataFrame.fields.filter((field) => field.config.links?.length && field.config.links?.length > 0) || [];
-    const orderedDataFrameFields = [
-      ...fieldsWithLinks,
-      ...dataFrame.fields.filter((field) => !fieldsWithLinks.includes(field)),
-    ];
+    // Show only configured labels and in the same order
+    const exemplarLabels = options?.exemplars?.labels;
+    let orderedDataFrameFields = [];
+    if (!exemplarLabels || exemplarLabels?.length === 0) {
+      const fieldsWithLinks =
+        dataFrame.fields.filter((field) => field.config.links?.length && field.config.links?.length > 0) || [];
+      orderedDataFrameFields = [
+        ...fieldsWithLinks,
+        ...dataFrame.fields.filter((field) => !fieldsWithLinks.includes(field)),
+      ];
+    } else {
+      for (let i in exemplarLabels) {
+        const label = exemplarLabels[i];
+        const field = dataFrame.fields.find((field) => field.name === label);
```

Comment:
@itsmylife can exemplar labels be changed via an override etc? We have ran into issues with this before. It may be safest to change this out with `getFieldDisplayName(field)` vs just `field.name` as these can in theory be different (see https://github.com/grafana/grafana/pull/73332#issuecomment-1688340234)

Reply:
Exemplar labels cannot be changed via an override nor any other UI in the Prometheus ds. Agreed, `getFieldDisplayName(field)` is preferred because as you state, they can be different.